bootloader/grub_test.go

```diff
@@ -413,8 +413,18 @@ func (s *grubTestSuite) TestGrubExtractedRunKernelImageEnableKernel(c *C) {
 	// ensure that the symlink was put where we expect it
 	asset, err := os.Readlink(filepath.Join(s.grubDir(), "kernel.efi"))
 	c.Assert(err, IsNil)
-
 	c.Assert(asset, DeepEquals, filepath.Join("pc-kernel_1.snap", "kernel.efi"))
+
+	// create a new kernel snap and ensure that we can safely enable that one
+	// too
+	kernel2 := s.makeKernelAssetSnap(c, "pc-kernel_2.snap")
+	err = eg.EnableKernel(kernel2)
+	c.Assert(err, IsNil)
+
+	// ensure that the symlink was put where we expect it
```

Comment:
Testing the unlikely name conflict situation would be interesting, but if the atomic symlink update code is extracted it could be tested for that function instead.

Reply:
I think you're right and it's a good idea to test the unlikely intermediate name conflict situation. I'll add a test for this.

Reply:
This is now tested in this PR from @bboozzoo's PR